File: aiJ-room/src/main/java/com/xiyoufang/aij/room/hero/Hero.java
package com.xiyoufang.aij.room.hero;

import com.xiyoufang.aij.room.table.Table;

/**
 * Created by 席有芳 on 2018-12-20.
 * 玩家
 *
 * @author 席有芳
 */
public class Hero {

    /**
     * 显示用的ID
     */
    private String showId;
    /**
     * 玩家名称
     */
    private String userName;
    /**
     * 玩家唯一ID
     */
    private String userId;
    /**
     * 昵称
     */
    private String nickName;
    /**
     * 性别
     */
    private Integer gender;
    /**
     * 头像
     */
    private String avatar;
    /**
     * 分销商ID
     */
    private String distributorId;
    /**
     * 经度
     */
    private Double longitude;
    /**
     * 纬度
     */
    private Double latitude;
    /**
     * 地址
     */
    private String address;
    /**
     * ip
     */
    private String ip;
    /**
     * 认证状态
     */
    private String certStatus;
    /**
     * 机器人标识
     */
    private Integer android;

    /**
     * 桌子编号
     */
    private int tableNo = Table.INVALID_TABLE;

    public String getShowId() {
        return showId;
    }

    public void setShowId(String showId) {
        this.showId = showId;
    }

    public String getUserName() {
        return userName;
    }

    public void setUserName(String userName) {
        this.userName = userName;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }

    public int getTableNo() {
        return tableNo;
    }

    public void setTableNo(int tableNo) {
        this.tableNo = tableNo;
    }

    public void setNickName(String nickName) {
        this.nickName = nickName;
    }

    public String getNickName() {
        return nickName;
    }

    public void setGender(Integer gender) {
        this.gender = gender;
    }

    public Integer getGender() {
        return gender;
    }

    public void setAvatar(String avatar) {
        this.avatar = avatar;
    }

    public String getAvatar() {
        return avatar;
    }

    public void setLongitude(Double longitude) {
        this.longitude = longitude;
    }

    public Double getLongitude() {
        return longitude;
    }

    public void setLatitude(Double latitude) {
        this.latitude = latitude;
    }

    public Double getLatitude() {
        return latitude;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    public String getIp() {
        return ip;
    }

    public void setIp(String ip) {
        this.ip = ip;
    }

    public String getCertStatus() {
        return certStatus;
    }

    public void setCertStatus(String certStatus) {
        this.certStatus = certStatus;
    }

    public String getDistributorId() {
        return distributorId;
    }

    public void setDistributorId(String distributorId) {
        this.distributorId = distributorId;
    }

    public Integer getAndroid() {
        return android;
    }

    public void setAndroid(Integer android) {
        this.android = android;
    }
}

File: aiJ-client/assets/Script/hero/Hero.ts
/**
 * 玩家类
 */
import AppConfig from "../AppConfig";

export default class Hero {

    /**
     * 玩家名称
     */
    userName: string;
    /**
     * show ID
     */
    showId: string;
    /**
     * ID
     */
    userId: string;
    /**
     * 昵称
     */
    nickName: string;
    /**
     * 性别
     */
    gender: number;
    /**
     * 头像
     */
    avatar: string;
    /**
     * 代理
     */
    distributorId: string;
    /**
     * 地址
     */
    address: string;
    /**
     * 经度
     */
    longitude: number;
    /**
     * 纬度
     */
    latitude: number;
    /**
     * Ip地址
     */
    ip: string;
    /**
     * 实名认证
     */
    certStatus: string;

    /**
     *
     * @param userName
     * @param showId
     * @param userId
     * @param nickName
     * @param gender
     * @param avatar
     * @param distributorId
     * @param address
     * @param longitude
     * @param latitude
     * @param ip
     * @param certStatus
     */
    constructor(userName: string, showId: string, userId: string, nickName: string, gender: number, avatar: string, distributorId: string,
                address: string, longitude: number, latitude: number, ip: string, certStatus: string) {
        this.userName = userName;
        this.showId = showId;
        this.userId = userId;
        this.nickName = nickName;
        this.gender = gender;
        this.avatar = cc.sys.isBrowser ? AppConfig.AVATAR_PROXY_URL + "?url=" + encodeURIComponent(avatar) : avatar;
        this.distributorId = distributorId;
        this.address = address;
        this.longitude = longitude;
        this.latitude = latitude;
        this.ip = ip;
        this.certStatus = certStatus;
    }
}
